drivers/spi/spi_silabs_eusart.c

```diff
@@ -152,6 +212,314 @@ static int spi_silabs_eusart_configure(const struct device *dev, const struct sp
+				    size_t requested_transaction_size, bool is_tx)
+{
+	memset(new_blk_cfg, 0, sizeof(*new_blk_cfg));
+	size_t remaining_transaction_size = 0U;
+
+	/* Set-up source and destination address with increment behavior. */
+	if (is_tx) {
+		new_blk_cfg->dest_address = (uint32_t)&instance_config->base->TXDATA;
+		new_blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+		if (buffer) {
+			new_blk_cfg->source_address = (uint32_t)buffer;
+			new_blk_cfg->source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+		} else {
+			/* Null buffer pointer means sending dummy byte. */
+			new_blk_cfg->source_address = (uint32_t)&dummy_byte;
+			new_blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+		}
+	} else {
+		new_blk_cfg->source_address = (uint32_t)&instance_config->base->RXDATA;
+		new_blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+		if (buffer) {
+			new_blk_cfg->dest_address = (uint32_t)buffer;
+			new_blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+		} else {
+			/* Null buffer pointer means rx to null byte. */
+			new_blk_cfg->dest_address = (uint32_t)&empty_buffer;
+			new_blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+		}
+	}
+
+	/* Setup max transfer according to requested transaction size.
+	 * Will top if bigger than the maximum transfer size.
+	 */
+	if (requested_transaction_size <= SPI_DMA_MAX_DESCRIPTOR_TRANSFER_SIZE) {
+		new_blk_cfg->block_size = requested_transaction_size;
+		remaining_transaction_size = 0U;
+	} else {
+		new_blk_cfg->block_size = SPI_DMA_MAX_DESCRIPTOR_TRANSFER_SIZE;
+		remaining_transaction_size = requested_transaction_size - new_blk_cfg->block_size;
+	}
+
+	/* Next block ptr. */
+	if (remaining_transaction_size > 0U) {
+		new_blk_cfg->next_block = &(new_blk_cfg[1]);
+	} else {
+		new_blk_cfg->next_block = NULL;
+	}
+
+	/* Return how much of the requested transaction is still required to be processed. */
+	return remaining_transaction_size;
+}
+
+static void spi_silabs_build_dma_config(const struct device *dev,
+					struct dma_config *current_dma_config,
+					const struct dma_block_config *blk_cfg,
+					uint32_t block_count, bool is_tx)
+{
+	memset(current_dma_config, 0, sizeof(*current_dma_config));
+
+	if (is_tx) {
+		/* Memory to peripheral. */
+		current_dma_config->channel_direction = 0x02U;
+	} else {
+		/* Peripheral to memory. */
+		current_dma_config->channel_direction = 0x01U;
+	}
+
+	/* invoke callback at the end of transfer list. */
+	current_dma_config->complete_callback_en = 0x00U;
+
+	current_dma_config->source_data_size = 0x01U; /* 1 byte. */
+	current_dma_config->dest_data_size = 0x01U;   /* 1 byte. */
+
+	current_dma_config->source_burst_length = 0x01U;
+	current_dma_config->dest_burst_length = 0x01U;
+
+	current_dma_config->block_count = block_count;
+
+	current_dma_config->head_block = (struct dma_block_config *)blk_cfg;
+
+	current_dma_config->user_data = (void *)dev;
+
+	if (!is_tx) {
+		/* Only enable the end transaction callback. Only SPI master mode is supported. */
```

Comment:
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language
Should be `SPI controller mode`.